### DISCUSSION TITLE
CircleCI upload_to_testflight error iTMSTransporter

### DISCUSSION THREAD

**Discussion by eclecticjohny** [fastlane/fastlane#18638] [Q&A]
Locally our teams builds are working using Xcode 12.5 but after updating our CircleCI pipeline to also use Xcode 12(12.4) from Xcode 11(11.4) Fastlane is now erroring out at the upload_to_testflight command. The below is the output with --verbose flag when it gets to the upload step. Not sure what to do as relevant search results do not seem to be specific this problem.

`INFO [2021-05-03 17:54:41.87]: Ready to upload new build to TestFlight (App: 1436644653)...
DEBUG [2021-05-03 17:54:41.87]: App Platform (ios)
INFO [2021-05-03 17:54:43.33]: Wrote XML data to '/var/folders/6y/gy9gggt14379c_k39vwb50lc0000gn/T/d20210503-5868-slzsgr/1436644653.itmsp'
INFO [2021-05-03 17:54:43.33]: Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
INFO [2021-05-03 17:54:43.33]: Going to upload updated app to App Store Connect
INFO [2021-05-03 17:54:43.33]: This might take a few minutes. Please don't interrupt the script.
DEBUG [2021-05-03 17:54:43.49]: /Applications/Xcode-12.4.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/itms/java/bin/java -Djava.ext.dirs=/Applications/Xcode-12.4.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/itms/java/lib/ext -XX:NewSize=2m -Xms32m -Xmx1024m -Xms1024m -Djava.awt.headless=true -Dsun.net.http.retryPost=false -jar /Applications/Xcode-12.4.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/itms/lib/itmstransporter-launcher.jar -m upload -u redacted@cava.com -p YourPassword -f /var/folders/6y/gy9gggt14379c_k39vwb50lc0000gn/T/d20210503-5868-slzsgr/1436644653.itmsp -t DAV,Signiant -k 100000 2>&1
WARN [2021-05-03 17:54:43.75]: [iTMSTransporter] -Djava.ext.dirs=/Applications/Xcode-12.4.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/itms/java/lib/ext is not supported.  Use -classpath instead.

WARN [2021-05-03 17:54:43.75]: [iTMSTransporter] Error: Could not create the Java Virtual Machine.

WARN [2021-05-03 17:54:43.75]: [iTMSTransporter] Error: A fatal exception has occurred. Program will exit.

INFO [2021-05-03 17:54:43.75]: iTunes Transporter output above ^
ERROR [2021-05-03 17:54:43.75]: The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.
WARN [2021-05-03 17:54:43.78]: Lane Context:
INFO [2021-05-03 17:54:43.78]: {:PLATFORM_NAME=>:ios, :LANE_NAME=>"ios deploy", :ORIGINAL_DEFAULT_KEYCHAIN=>"\"/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db\""}
ERROR [2021-05-03 17:54:43.78]: Error uploading ipa file, for more information see above
INFO [2021-05-03 17:54:43.78]: Successfully generated documentation at path '/Users/distiller/project/packages/cava-mobile/fastlane/README.md'

+------+----------------------+-------------+
|             fastlane summary              |
+------+----------------------+-------------+
| Step | Action               | Time (in s) |
+------+----------------------+-------------+
| 1    | setup_circle_ci      | 0           |
| 💥   | upload_to_testflight | 7           |
+------+----------------------+-------------+

ERROR [2021-05-03 17:54:43.80]: fastlane finished with errors
bundler: failed to load command: fastlane (/Users/distiller/project/packages/cava-mobile/vendor/bundle/bin/fastlane)
FastlaneCore::Interface::FastlaneError: [!] Error uploading ipa file, for more information see above
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/pilot/lib/pilot/build_manager.rb:37:in `upload'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:261:in `block (2 levels) in execute_action'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:253:in `block in execute_action'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
  Fastfile:189:in `block (2 levels) in parsing_binding'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/gems/fastlane-2.144.0/bin/fastlane:23:in `<top (required)>'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/bin/fastlane:23:in `load'
  /Users/distiller/project/packages/cava-mobile/vendor/bundle/bin/fastlane:23:in `<top (required)>'

Exited with code exit status 1
CircleCI received exit code 1`

**Comment by mathaeus** [comment]
@eclecticjohny it seems you're using quite an old `fastlane` version (`2.144.0`), so maybe updating to the latest version helps.

We are also using `circleCI` and are uploading builds to TestFlight via `fastlane` without any issues.
